src/test/java/com/yahoo/bullet/querying/FilterOperationsTest.java

```diff
@@ -591,4 +595,14 @@ public void testNested() {
         Assert.assertFalse(FilterOperations.perform(recordF, clause));
         Assert.assertTrue(FilterOperations.perform(recordG, clause));
     }
+
+    @Test
+    public void testCompareToFields() {
```

Comment:
Can we add a test for what happens when an ObjectFilterClause is being tested and the cast cannot be done (the RunTimeException is thrown)?

Reply:
It will be ignored. Added a test.